cranelift/codegen/meta/src/shared/legalize.rs

```diff
@@ -213,6 +216,30 @@ pub(crate) fn define(insts: &InstructionGroup, imm: &Immediates) -> TransformGro
     // embedded as part of arguments), so use a custom legalization for now.
     narrow.custom_legalize(iconst, "narrow_iconst");
 
+    narrow.legalize(
+        def!(istore32.I64(flags, a, ptr, offset)),
+        vec![
+            def!((al, ah) = isplit(a)),
+            def!(store.I32(flags, al, ptr, offset)),
+        ],
+    );
```

Comment:
In this legalization we are trying to store the lower 32 bits of an `I64`. It makes sense to me that this would be in `narrow`.